gbpservice/nfp/configurator/agents/generic_config.py

```diff
@@ -0,0 +1,468 @@
+        """Enqueues event for worker to process clear routes request.
+
+        :param context: RPC context
+        :param kwargs: RPC Request data
+
+        Returns: None
+
+        """
+
+        arg_dict = {'context': context,
+                    'kwargs': kwargs}
+        ev = self._sc.new_event(id=gen_cfg_const.EVENT_CLEAR_ROUTES,
+                                data=arg_dict)
+        self._sc.post_event(ev)
+
+    def configure_healthmonitor(self, context, kwargs):
+        """Enqueues event for worker to process configure healthmonitor request.
+
+        :param context: RPC context
+        :param kwargs: RPC Request data
+
+        Returns: None
+
+        """
+
+        kwargs['fail_count'] = 0
+        arg_dict = {'context': context,
+                    'kwargs': kwargs}
+        ev = self._sc.new_event(id=gen_cfg_const.EVENT_CONFIGURE_HEALTHMONITOR,
+                                data=arg_dict, key=kwargs['vmid'])
+        self._sc.post_event(ev)
+
+    def clear_healthmonitor(self, context, kwargs):
+        """Enqueues event for worker to process clear healthmonitor request.
+
+        :param context: RPC context
+        :param kwargs: RPC Request data
+
+        Returns: None
+
+        """
+
+        arg_dict = {'context': context,
+                    'kwargs': kwargs}
+        ev = self._sc.new_event(id=gen_cfg_const.EVENT_CLEAR_HEALTHMONITOR,
+                                data=arg_dict, key=kwargs['vmid'])
+        self._sc.post_event(ev)
+
+
+"""Implements event handlers and their helper methods.
+
+Object of this class is registered with the event class of core service
+controller. Based on the event key, handle_event method of this class is
+invoked by core service controller.
+
+"""
+
+
+class GenericConfigEventHandler(agent_base.AgentBaseEventHandler,
+                                nfp_poll.PollEventDesc):
+    def __init__(self, sc, drivers, rpcmgr, nqueue):
+        super(GenericConfigEventHandler, self).__init__(
+                                        sc, drivers, rpcmgr, nqueue)
+        self.sc = sc
+
+    def _get_driver(self, service_type):
+        """Retrieves service driver object based on service type input.
+
+        Currently, service drivers are identified with service type. Support
+        for single driver per service type is provided. When multi-vendor
+        support is going to be provided, the driver should be selected based
+        on both service type and vendor name.
+
+        :param service_type: Service type - firewall/vpn/loadbalancer
+
+        Returns: Service driver instance
+
+        """
+
+        return self.drivers[service_type]()
+
+    def _notification(self, data):
+        LOG.info(" Sending notification=%s " % (data))
+        event = self.sc.new_event(
+            id='NOTIFICATION_EVENT', key='NOTIFICATION_EVENT', data=data)
+        self.sc.poll_event(event)
+
+    def handle_event(self, ev):
+        """Processes the generated events in worker context.
```

Comment:
- Can event processing this method and _process_event be broken up into different methods?
- Can you make use of a table as in lifecycle_manager/modules